lib/token.h

```diff
@@ -187,6 +187,30 @@ class CPPCHECKLIB Token {
     explicit Token(TokensFrontBack *tokensFrontBack = nullptr);
     ~Token();
 
+    struct TokenEnumeration
+    {
+        Token const* mt;
+        TokenEnumeration(Token const* t) : mt(t) {}
+
+        struct TokenIterator
+        {
+            const Token* mt;
+            TokenIterator(const Token* t) : mt(t) {}
```

Comment:
This needs a default constructor. Iterators are required to be default constructible(see [here](https://en.cppreference.com/w/cpp/named_req/ForwardIterator)).